akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcOffsetStore.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+import java.sql.Connection
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Instant
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.projection.MergeableOffset
+import akka.projection.ProjectionId
+import akka.projection.internal.OffsetSerialization
+import akka.projection.internal.OffsetSerialization.SingleOffset
+import akka.projection.internal.OffsetSerialization.toStorageRepresentation
+import akka.projection.internal.OffsetSerialization.fromStorageRepresentation
+import akka.projection.jdbc.javadsl.JdbcSession
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] class JdbcOffsetStore(settings: JdbcSettings, clock: Clock) {
+
+  def createIfNotExists[S <: JdbcSession](jdbcSessionFactory: () => S)(implicit ec: ExecutionContext): Future[Done] = {
+    JdbcSession
+      .runBlockingDBIO(jdbcSessionFactory) { sess =>
+        sess.withConnection { conn =>
+          val stmt = conn.createStatement()
+          stmt.execute(settings.dialect.createTableStatement)
+          stmt.execute(settings.dialect.alterTableStatement)
+          Done
+        }
+      }
+  }
+
+  def clearOffset(conn: Connection, projectionId: ProjectionId) = {
+    val stmt = conn.prepareStatement(settings.dialect.clearOffsetStatement)
+    stmt.setString(1, projectionId.name)
+    stmt.setString(2, projectionId.key)
+    stmt.executeUpdate()
```

Comment:
I think it's important to close PreparedStatement (and also ResultSet)m but understand that it's not part of the first wip

Reply:
For info, I added a `tryWithResource` method that takes a stmt, passes it to a lambda and close it at the end. 

I'm not closing the result set because they are closed automatically when you close the statement. Since we close it immediately after using the result set, I think it's not necessary. That free us to add extra lines of code.

Reply:
now tryWithResources can be used with any AutoCloseable.